packages/ui/package.json

```diff
@@ -42,9 +42,17 @@
   "dependencies": {
     "lodash": "4.17.21",
     "style-dictionary": "3.7.0",
-    "xstate": "^4.33.6",
     "tslib": "2.4.0"
   },
+  "peerDependencies": {
+    "aws-amplify": "3.x.x || 4.x.x",
+    "xstate": "^4.33.6"
+  },
+  "peerDependenciesMeta": {
+    "xstate": {
+      "optional": true
```

Comment:
QQ, why is this optional?

Reply:
I think this keeps NPM 8+ from automatically downloading it.

Reply:
Gotcha, it makes sense to do this then

Reply:
Also, we use the UI package in 2 places: once in the _react-core_ package and once in the _react_ package. The _react-core_ is the only one that uses the `xstate` functionality, while _react_ uses the styling aspects. Thus, it makes sense to treat `xstate` as an optional peer dependency since it isn't used by `@aws-amplify/ui-react`.